src/lightning_app/core/app.py

```diff
@@ -545,6 +629,7 @@ def load_state_dict_from_checkpoint_dir(
         self.load_state_dict(state)
 
     def _dump_checkpoint(self) -> Optional[str]:
+        # TODO: revisit this logic.
```

Comment:
I think you meant to say that this method should be removed once the checkpointing is no longer experimental